From 8c6767d80669d7097a987c04e1907583eda859e3 Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Thu, 2 May 2013 16:55:00 +0800
Subject: [PORT FROM MAIN] libstagefright: avoid crash by checking the configuration vserion

BZ: 94702

The value of configuration version in avcc box should be 1 but some
malformed clip contains other value and causes mediaserver crash.

Change-Id: Id820b13534c161b8d95580d864e084ea131f7a27
Orig-Change-Id: I4ecd889efbb883efed2a135cbe0bf0e429901ebd
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 4a5471f..b1561a2 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -1581,6 +1581,10 @@ status_t MPEG4Extractor::parseChunk(off64_t *offset, int depth) {
                 return ERROR_IO;
             }
 
+            // configurationVersion should be 1
+            if (buffer->data()[0] != 1)
+                return ERROR_MALFORMED;
+
             mLastTrack->meta->setData(
                     kKeyAVCC, kTypeAVCC, buffer->data(), chunk_data_size);
 
-- 
1.7.4.1

